src/integrations/admin/indexing-integration.php

```diff
@@ -0,0 +1,233 @@
+	 *
+	 * @var Indexable_General_Indexation_Action
+	 */
+	protected $general_indexation;
+
+	/**
+	 * Represented the indexation completed action.
+	 *
+	 * @var Indexable_Complete_Indexation_Action
+	 */
+	protected $complete_indexation_action;
+
+	/**
+	 * The total amount of unindexed objects.
+	 *
+	 * @var int
+	 */
+	protected $total_unindexed;
+
+	/**
+	 * Represents the admin asset manager.
+	 *
+	 * @var WPSEO_Admin_Asset_Manager
+	 */
+	protected $asset_manager;
+
+	/**
+	 * The indexable helper.
+	 *
+	 * @var Indexable_Helper
+	 */
+	protected $indexable_helper;
+
+	/**
+	 * Returns the conditionals based on which this integration should be active.
+	 *
+	 * @return array The array of conditionals.
+	 */
+	public static function get_conditionals() {
+		return [
+			Yoast_Tools_Page_Conditional::class,
+			Migrations_Conditional::class,
+		];
+	}
+
+	/**
+	 * Indexing_Integration constructor.
+	 *
+	 * @param Indexable_Post_Indexation_Action              $post_indexation              The post indexing action.
+	 * @param Indexable_Term_Indexation_Action              $term_indexation              The term indexing action.
+	 * @param Indexable_Post_Type_Archive_Indexation_Action $post_type_archive_indexation The post type archive indexing action.
+	 * @param Indexable_General_Indexation_Action           $general_indexation           The general indexing action.
+	 * @param Indexable_Complete_Indexation_Action          $complete_indexation_action   The complete indexing action.
+	 * @param WPSEO_Admin_Asset_Manager                     $asset_manager                The admin asset manager.
+	 * @param Indexable_Helper                              $indexable_helper             The indexable helper.
+	 */
+	public function __construct(
+		Indexable_Post_Indexation_Action $post_indexation,
+		Indexable_Term_Indexation_Action $term_indexation,
+		Indexable_Post_Type_Archive_Indexation_Action $post_type_archive_indexation,
+		Indexable_General_Indexation_Action $general_indexation,
+		Indexable_Complete_Indexation_Action $complete_indexation_action,
+		WPSEO_Admin_Asset_Manager $asset_manager,
+		Indexable_Helper $indexable_helper
+	) {
+		$this->post_indexation              = $post_indexation;
+		$this->term_indexation              = $term_indexation;
+		$this->post_type_archive_indexation = $post_type_archive_indexation;
+		$this->general_indexation           = $general_indexation;
+		$this->complete_indexation_action   = $complete_indexation_action;
+		$this->asset_manager                = $asset_manager;
+		$this->indexable_helper             = $indexable_helper;
+	}
+
+	/**
+	 * Register hooks.
+	 */
+	public function register_hooks() {
+		\add_action( 'wpseo_tools_overview_list_items', [ $this, 'render_indexing_list_item' ], 10 );
+		\add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts' ], 10 );
+	}
+
+	/**
+	 * Enqueues the required scripts.
+	 *
+	 * @return void
+	 */
+	public function enqueue_scripts() {
```

Comment:
In #16049, [in this commit](https://github.com/Yoast/wordpress-seo/pull/16049/commits/b4d4a46d7f17b14fecdfe09e544da3192b4d2937) the first thing we do in `enqueue_scripts()` is checking whether or not the indexables should be indexed. Is there a reason this check is not included in the new integration?

Reply:
Good question 🙂. 
The React component is now responsible for showing the 'Everything is correctly indexed' alert. This means that the JavaScript, its data and all the styling should be loaded.